### PR TITLE
Unify examples structure

### PR DESCRIPTION
**Pull request by spydon**
# Description

Unifies the structure or the examples and organizes them a bit.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Comment by st-pasha** [comment]
Can we add a rule that the file name should match the name of the main class? That is, `MoveEffectExample` to live in `move_effect_example.dart`.

This would have several benefits:
  - file names more standardized (`move_effect.dart` main class, `move_effect_test.dart` the test class, `move_effect_example.dart` the example class);
  - easier to find the example of interest (if I know that the name of the file will be `move_effect_example`, it's easier to use file search to open that file than navigating the file tree);
  - easier to work with individual examples (it is common to work on a class and on the example for that class at the same time, so if both files have same names it gets quite confusing).

File: STYLEGUIDE.md [hunk 1]
```diff
@@ -150,3 +150,13 @@ Avoid documentation that just repeats the obvious. For example, `void doStuff()`
 
 You should use `[]` (brackets) to link dartdoc elements that can be referenced on the same file.
 Also, consider adding a "See also" section to element documentation.
+
+## Examples
+
+There are a few things to consider when adding an example to the examples directory:
+
+ - Put your example in a fitting section.
+ - Always add a description named `description` in the game object (if you have one).
```
Comment (erickzanardo) [inline STYLEGUIDE.md:159]: I wonder if we shouldn't make this required?

File: STYLEGUIDE.md [hunk 2]
```diff
@@ -150,3 +150,13 @@ Avoid documentation that just repeats the obvious. For example, `void doStuff()`
...
+ - Always have the game class (if there exists one) in the top of your examples file.
```
Comment (erickzanardo) [inline STYLEGUIDE.md:160]: Maybe also mention the requirement of the source file?

File: examples/lib/stories/rendering/isometric_tile_map.dart [hunk 3]
```diff
@@ -7,45 +7,29 @@ import 'package:flame/input.dart';
 import 'package:flame/sprite.dart';
 import 'package:flutter/material.dart' hide Image;
 
-const x = 500.0;
-const y = 500.0;
-final topLeft = Vector2(x, y);
+class IsometricTileMapExample extends FlameGame with MouseMovementDetector {
```
Comment (luanpotter) [inline examples/lib/stories/rendering/isometric_tile_map.dart:10]: I was going to add a tiled example, that is why I created tile_maps :P 
but back how we were doing it then I was unable to add a flame_x package as dep of this package

**Comment by spydon** [comment]
> Can we add a rule that the file name should match the name of the main class? That is, `MoveEffectExample` to live in `move_effect_example.dart`.
> 
> This would have several benefits:
>   - file names more standardized (`move_effect.dart` main class, `move_effect_test.dart` the test class, `move_effect_example.dart` the example class);
>   - easier to find the example of interest (if I know that the name of the file will be `move_effect_example`, it's easier to use file search to open that file than navigating the file tree);
>   - easier to work with individual examples (it is common to work on a class and on the example for that class at the same time, so if both files have same names it gets quite confusing).

They are not organized like that currently, several example are overlapping the same features, but sounds like a good idea for another refactor.

File: STYLEGUIDE.md (hunk 1 above)
Comment (spydon) [reply STYLEGUIDE.md:159]: Sounds like a good idea to me!

File: STYLEGUIDE.md (hunk 2 above)
Comment (spydon) [reply STYLEGUIDE.md:162]: Done!

**Review by erickzanardo — APPROVED** [review]

File: examples/lib/stories/rendering/isometric_tile_map.dart (hunk 3 above)
Comment (spydon) [reply examples/lib/stories/rendering/isometric_tile_map.dart]: The `flame_tiled` example should be in the `example` folder of the package now? :)

Comment (luanpotter) [reply examples/lib/stories/rendering/isometric_tile_map.dart]: I think each package has its own `example` folder with a simple example of that package only, but I wanted to include more examples for all packages and their interactions on the examples app

Comment (erickzanardo) [reply examples/lib/stories/rendering/isometric_tile_map.dart]: I second this idea.

The individual examples are not so accessible, would be nice that they were bundled inside our Dashbook too.

Comment (spydon) [reply examples/lib/stories/rendering/isometric_tile_map.dart]: Sounds good! But we can bring back this section when there are more examples then one then?

**Pull request merged**